### PR TITLE
Fix/mac omp

### PR DESCRIPTION
**Pull request by drewitz**
needed to find omp on mac

### CODE REVIEW

**Comment by rgugliel** [comment]
Hey Simon!
Thanks for the PR. I'm still a bit unsure with these changes, since the current code used to work on macOS (or was it the old OSX) and I'd like to be sure we don't introduce regression.

Will have a look online about the issue and come back here.

Cheers,

Raf

**Comment by drewitz** [comment]
> the current code used to work on macOS

Did it? (Not that I remember.)
I thought the missing omp was the reason CoxIter is running slower on a mac. You also didn't include omp in the compile instructions in the documentation.
If you don't install omp, it's still compiling and running perfectly fine.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.4)
+cmake_minimum_required(VERSION 3.12)
```
Comment (rgugliel) [inline CMakeLists.txt:1]: Do we really need that?

Comment (drewitz) [reply CMakeLists.txt:1]: Yes. This is the line which makes the magic happen:

with VERSION 2.4:
```
-- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES) 
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND) 
```
with VERSION 3.12:
```
-- Found OpenMP_C: -Xclang -fopenmp (found version "3.1") 
-- Found OpenMP_CXX: -Xclang -fopenmp (found version "3.1") 
-- Found OpenMP: TRUE (found version "3.1")
```

(This is from two runs on Catalina. One after the other with a clean build directory, the only difference being the line you asked about.)

I'm not sure anymore where I found the solution, but it's mentioned in a stackoverflow answer here:
> For others seeing this in the future, update to CMake 3.12 if it has been released at the time of reading.

https://stackoverflow.com/questions/49738961/cmake-support-for-openmp-on-macos-high-sierra/50313962

Comment (rgugliel) [reply CMakeLists.txt:1]: Ok, thanks!

**Pull request merged**